include/pybind11/chrono.h

```diff
@@ -166,7 +166,7 @@ class type_caster<std::chrono::time_point<std::chrono::system_clock, Duration>>
             return false;
         }
 
-        value = time_point_cast<Duration>(system_clock::from_time_t(std::mktime(&cal)) + msecs);
+        value = time_point_cast<Duration>(system_clock::from_time_t(timegm(&cal)) + msecs);
```

Comment:
`timegm` isn't available on windows, so this line needs to change to something like:

```c++
#if defined(_MSC_VER)
    time_t tt = _mkgmtime(&cal);
#else
    time_t tt = timegm(&cal);
#endif

    value = time_point_cast<Duration>(system_clock::from_time_t(tt) + msecs);
```

Reply:
As @Skylion007 pointed out before, we don't have to go to UTC to solve the roundtrip problem: https://github.com/pybind/pybind11/pull/3805#discussion_r830116635
I think it's better to keep using the local time zone, but make proper use of the `fold` feature.

Reply:
with that solution would `std::system_clock` time points go into python as time aware (UTC) datetime objects, and python datetime objects go into C++ as UTC?

C++ system_clock is Unix Time (so UTC modulo leap seconds and such), so the time zone conversions are a little surprising.

Reply:
I believe python datetime objects should be in local time in both? I am not sure we are actually attaching the time zone to the Python objects (we should but it doesn't seem straight forward using the raw C - API). 

If it's too tricky, we might want to call the Python datetime function directly if we need to.